weather_mv/loader_pipeline/regrid.py

```diff
@@ -52,16 +188,28 @@ def add_parser_arguments(cls, subparser: argparse.ArgumentParser) -> None:
                                     """Will default to '{"grid": [0.25, 0.25]}'.""")
         subparser.add_argument('--to_netcdf', action='store_true', default=False,
                                help='Write output file in NetCDF via XArray. Default: off')
+        subparser.add_argument('-zi', '--zarr_input_chunks', type=json.loads, default='{}',
+                               help='When reading a Zarr, break up the data into chunks. Takes a JSON string.')
+        subparser.add_argument('-zo', '--zarr_output_chunks', type=json.loads, default='{}',
+                               help='When writing a Zarr, write the data with chunks. Takes a JSON string.')
 
     @classmethod
     def validate_arguments(cls, known_args: argparse.Namespace, pipeline_options: t.List[str]) -> None:
-        pass
+        if known_args.zarr and known_args.to_netcdf:
+            raise ValueError('only Zarr-to-Zarr regridding is allowed!')
+
+        if not known_args.zarr and (known_args.zarr_input_chunks or known_args.zarr_output_chunks):
+            raise ValueError('chunks can only be set when input URI is a Zarr.')
 
     def target_from(self, uri: str) -> str:
         """Create the target path from the input URI.
 
-        Will change the extension if the target is a NetCDF.
+        In the case of Zarr, the output will be treated like a valid path.
```

Comment:
This seems like strange and possibly unexpected behavior.

Reply:
The output Zarr path just needs to be a directory, otherwise it could take any shape. I agree that there is a footgun here. I've added a user warning to the validation step to nudge users to user more standard zarr formatting.